lib/rouster.rb

```diff
@@ -292,60 +293,87 @@ def inspect
   #
   # runs a command inside the Vagrant VM
   #
-  # returns output (STDOUT and STDERR) from command run, sets @exitcode
-  # currently determines exitcode by tacking a 'echo $?' onto the command being run, which is then parsed out before returning
+  # returns output (STDOUT and STDERR) from command run, pushes to self.ssh_exitcode
   #
   # parameters
   # * <command> - the command to run (sudo will be prepended if specified in object instantiation)
   # * [expected_exitcode] - allows for non-0 exit codes to be returned without requiring exception handling
-  def run(command, expected_exitcode=[0])
+  def run( command, expected_exitcode = 0, sudo = self.uses_sudo? )
```

Comment:
can we update the input parameters to include `sudo`? also, the context for `<command>` is correct, we still sudo if that was in object instantiation, and that is default, but as the parameter overrides it, i think we should remove that bit